packages/clerk-js/src/ui/utils/phoneUtils.ts

```diff
@@ -115,3 +116,34 @@ export function stringToFormattedPhoneString(str: string): string {
   const parsed = parsePhoneString(str);
   return `+${parsed.code} ${formatPhoneNumber(parsed.number, parsed.pattern, parsed.code)}`;
 }
+
+export const sortByPriority = (a: CountryEntry, b: CountryEntry) => {
+  if (a.priority > b.priority) {
+    return -1;
+  }
+  if (a.priority < b.priority) {
+    return 1;
+  }
+  return 0;
+};
+
+export function getLongestValidCountryCode(value: string) {
+  let result = '';
+  let countries = null;
+  const numberValue = value.replace('+', '');
+  const countryGroup = [];
+
+  for (let i = 4; i > 0; i--) {
+    const code = numberValue.slice(0, i);
+    countries = [...IsoToCountryMap.values()].filter(o => o.code === code);
+
+    if (countries.length) {
+      countryGroup.push(countries);
+    }
+  }
+
+  const selectedCountry = countryGroup.flat().sort(sortByPriority)[0];
+  result = numberValue.slice(selectedCountry?.code.length, value.length);
+
+  return { result, selectedCountry };
```

Comment:
Whats the difference between `result` and `selectedCountry`? Can we make result more semantic, otherwise we'd have to look at the functions implementation everytime

Reply:
@nikos this is the phone number value without the `+`. I will rename